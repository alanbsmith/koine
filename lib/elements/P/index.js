import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import fontWeights from '../../modifiers/fontWeights';

const customModifiers = {
  italic: () => 'font-style: italic;',
  extraSmall: () => 'font-size: 12px;',
  small: () => 'font-size: 14px;',
  large: () => 'font-size: 18px;',
};

const modifierConfig = { ...fontWeights, ...customModifiers };

const P = styled.p`
  font-size: 16px;
  margin: 0;
  ${applyStyleModifiers(modifierConfig)}
`;

export default P;
