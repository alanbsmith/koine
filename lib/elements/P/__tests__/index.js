import React from 'react';
import P from '../index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Paragraph Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <P>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the capitalize modifier', () => {
    const tree = renderer.create(
      <P modifiers={['capitalize']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the italicize modifier', () => {
    const tree = renderer.create(
      <P modifiers={['italicize']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the lowercase modifier', () => {
    const tree = renderer.create(
      <P modifiers={['lowercase']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the underline modifier', () => {
    const tree = renderer.create(
      <P modifiers={['underline']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the uppercase modifier', () => {
    const tree = renderer.create(
      <P modifiers={['uppercase']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the extraSmall modifier', () => {
    const tree = renderer.create(
      <P modifiers={['extraSmall']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the large modifier', () => {
    const tree = renderer.create(
      <P modifiers={['large']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the small modifier', () => {
    const tree = renderer.create(
      <P modifiers={['small']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the lightWeight modifier', () => {
    const tree = renderWithTheme(
      <P modifiers={['lightWeight']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the normalWeight modifier', () => {
    const tree = renderWithTheme(
      <P modifiers={['normalWeight']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the semiBoldWeight modifier', () => {
    const tree = renderWithTheme(
      <P modifiers={['semiBoldWeight']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the boldWeight modifier', () => {
    const tree = renderWithTheme(
      <P modifiers={['boldWeight']}>A paragraph</P>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
