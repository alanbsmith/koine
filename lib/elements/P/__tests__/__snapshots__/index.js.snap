// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Element adds the boldWeight modifier 1`] = `
<p
  className="sc-bdVaJa fBmCNT"
>
  A paragraph
</p>
`;

exports[`Header Element adds the extraSmall modifier 1`] = `
<p
  className="sc-bdVaJa gYjnOM"
>
  A paragraph
</p>
`;

exports[`Header Element adds the italic modifier 1`] = `
<p
  className="sc-bdVaJa dSDadT"
>
  A paragraph
</p>
`;

exports[`Header Element adds the large modifier 1`] = `
<p
  className="sc-bdVaJa iHluKy"
>
  A paragraph
</p>
`;

exports[`Header Element adds the small modifier 1`] = `
<p
  className="sc-bdVaJa iRqVHv"
>
  A paragraph
</p>
`;

exports[`Header Element renders correctly 1`] = `
<p
  className="sc-bdVaJa fNqJPA"
>
  Header Element
</p>
`;
